### PR TITLE
Fix git resource to use revision attribute

### PR DESCRIPTION
**Pull request by iankronquist**
https://github.com/chef/chef/pull/3289#issuecomment-113790597

This is probably the result of a docs bug in chef.

@tschuy r?

### CODE REVIEW

**Comment by tschuy** [comment]
Update the tests and I'll take a look on monday.

**Comment by tschuy** [comment]
Looks good to me. I'm running the integration tests, but I think they'll pass. I'll merge when they do.

**Comment by tschuy** [comment]
I've pushed some integration tests that check the revision checkout. My ubuntu converges failed for unreleated reasons and I've got to go.

**Comment by tschuy** [comment]
@osuosl-cookbooks/Chefs @osuosl-cookbooks/chefs

**Comment by jordane** [comment]
@osuosl-cookbooks/chefs

**Comment by tschuy** [comment]
I think this is ready to merge. sounds like @ElijahCaine is looking to use it, so it'd be nice to get this merged into master soon.

File: .kitchen.cloud.yml [hunk 1]
```diff
@@ -44,6 +44,9 @@ platforms:
   driver_config:
     username: ubuntu
     image_ref: "Ubuntu 14.04"
+  run_list:
+   - recipe[apt]
+   - recipe[python-webapp-test::default]
```
Comment (ramereth) [inline .kitchen.cloud.yml]: Why do you have all the `recipe[python-webapp-test::default]` included under each platform? Shouldn't that be covered by the default suite?

**Comment by ramereth** [comment]
I'm getting a lot of output like the following when running the tests:

```
No backend type is specified. Fall back to :exec type.
```

File: .kitchen.cloud.yml (hunk 1 above)
Comment (tschuy) [inline .kitchen.cloud.yml]: Oops.

**Comment by ramereth** [comment]
:+1:

**Comment by pop** [comment]
I ran the tests. Nothing blew up. :+1:

**Pull request merged**